Packs/Legacy/Scripts/FindSimilarIncidentsV2/find_similar_incidents_test.py

```diff
@@ -182,35 +209,66 @@ def test_similar_context(mocker):
     assert result['EntryContext']['similarIncidentList'][0]['rawId'] == 3
     assert result['EntryContext']['similarIncidentList'][1]['rawId'] == 2
 
-    args.update({'similarIncidentFields': 'name', 'similarContextKeys': 'dictListValue'})
+
+def test_similar_context_simple_list_value(mocker):
+    args = dict(default_args)
+    args.update({'similarIncidentFields': 'name', 'similarContextKeys': 'simpleListValue'})
+
+    mocker.patch.object(demisto, 'args', return_value=args)
+    mocker.patch.object(demisto, 'incidents', return_value=[incident1])
+    mocker.patch.object(demisto, 'executeCommand', side_effect=execute_command)
+    mocker.patch.object(demisto, 'context', return_value=context1)
+    mocker.patch.object(demisto, 'dt', side_effect=dt_res)
+
     result = main()
-    assert len(result['EntryContext']['similarIncidentList']) == 1
-    assert result['EntryContext']['similarIncidentList'][0]['rawId'] == 2
+    assert len(result['EntryContext']['similarIncidentList']) == 2
+    assert result['EntryContext']['similarIncidentList'][0]['rawId'] == 3
+    assert result['EntryContext']['similarIncidentList'][1]['rawId'] == 2
+
+
+def test_similar_context_missing_key(mocker):
+    args = dict(default_args)
+    args.update({'skipMissingValues': 'no', 'similarContextKeys': 'missingKey'})
+
+    mocker.patch.object(demisto, 'args', return_value=args)
+    mocker.patch.object(demisto, 'incidents', return_value=[incident1])
+    mocker.patch.object(demisto, 'executeCommand', side_effect=execute_command)
+    mocker.patch.object(demisto, 'context', return_value=context1)
+    mocker.patch.object(demisto, 'dt', side_effect=dt_res)
+
+    with pytest.raises(ValueError, match="Error: Missing context key for incident: missingKey"):
+        main()
 
 
-def dt_res(context, key):
-    if key == 'simpleValue':
-        if context is context2:
-            return 'simple'
+def dt_res(context, keys_to_search):
+
+    keys_list = keys_to_search.split('.')
+    context_key_value = None
+    for key in keys_list:
+        if context_key_value:
+            if isinstance(context_key_value, list):
+                list_value = []
+                for value in context_key_value:
+                    if isinstance(context_key_value, dict):
+                        if value.get(key):
+                            list_value.append(value[key])
+                    else:
+                        list_value.append(value)
+                context = context_key_value = list_value
```

Comment:
why this?
why not just `context_key_value = list_value`

Reply:
you are right
I will change it